### DISCUSSION TITLE
DSDPlus - Digital Decoder

### DISCUSSION THREAD

**Discussion by Djonget** [chuot/rdio-scanner#244] [General]
DSDPlus Fast Lane 2.71 (https://ufile.io/fuz26on6) Public versions
D-Star/NXDN4800/NXDN9600/DMR/MotoTRBO/X2-TDMA/ProVoice/P25/dPMR/IDAS/CAP+/Con+/TIII/DCDM/XPT (For the latest version to developers in the mail)

Good afternoon, I use a powerful decoder of digital signals, knows how to save audio files in WAW and MP3 format, creates folders by format: VC-Recorder\20220721\Files.mp3
**********************\20220722\Files.mp3
**********************\20220723\Files.mp3

It is not possible to configure Dirwatch, I ask for the relics in the setting.
File format:

140951_001_P25_BEE00.1C3-3.16_GC_102_180698.mp3
141020_007_P25_BEE00.5C5-5.15_PC_12_1702698.mp3
120951_009_P25_BEE00.7C7-1.16_GC_202_13698.mp3

Time _ ? _ Protocol _ NetworkID-Site _ ? _ IDgroup _ IDradio(RadioAlias).mp3

**Comment by chuot** [comment]
Is the filename format configurable on the DSDPlus side so that it can be compatible with the current dirwatch meta tags?

I could create more meta tags but since I don't use DSDPlus, and don't have access to DSDPlus fastlane, I will need more details/specs on all the filename formats that DSDPlus fastlane can generate before I can modify Rdio Scanner. Maybe the solution is not new meta tags but rather a new dirwatch.type=dsdplus. I also assume that the filenames will change depending on the type of system (nxdn, dmr, ...).

**Reply by noamlivne** [reply]
@chuot and @Djonget. Maybe this can help:

Here are examples from DSDPlus Fastlane.

Note that the recordings are nested under each relevant and current date...
So \DSDPlus2P368\1R-Record#6\20220809 
etc... So each date creates a new subfolder...
The 1R-Record## is a recorder name, per dongle (optional).

I enabled the Per-Call Recording (wav or mp3) in the Output menu of DSDPlus and Added Talkgroups and Aliases per call filename.

DSDPlus Fastlane is much more sensitive than SDRTrunk - so it can be useful for picking up stations that SDRTrunk does not pick up + it can do NXDN...
(DSDPlus cannot record NFM or AM, but it can record non-trunked digital transmissions during a Scan)

NXDN examples:
152142_001_NEXEDGE48(CB)__RAN5_GC_0_235.wav
152144_002_NEXEDGE48(CB)__RAN5_GC_0_97[---].wav
The first 6 numbers is the local time
The 3 next numbers I have no idea
The next long field is the name of the "service method", NXDN CB
Then the RAN number appears (which is part of NXDN)
Then GC is group call I would guess. I guess that 0 is the Talkgroup which is non-existent in NXDN.
And the last number and [xxx] is the UID/RID

DMR examples:
153120_001_DMR(BS)_1-899_DCC2_Slot1_GC_750[Ram_Muni]_52[_________].wav
153128_001_DMR(BS)_1-899_DCC2_Slot1_GC_750[Ram_Muni]_1.wav
So "TIME"_"unknown 3 numbers"_"Service method"_"Fictional Network numbers I gave it"_"Color code"_"Slot number"_"Group call"_"Talk group[Talkgroup ID I set up]"_"RID[with or without name]".wav 

P25 examples:
152124_000_P25(BS)_BEE00.37D-1.5_GC_1285[Reh_5016]_3150006.wav
152129_001_P25(BS)_BEE00.37D-1.5_GC_1281[Ris_5014]_3150005.wav
So TIME"_"unknown 3 numbers"_"Service method"_"WACN.SYSID-SiteID"_"Group call"_Talk group[Talk group name I set up]"_RID[with or without name]".wav

DMR Simplex/Repeater Input (I am not sure SDRTrunk can do that):
151405_014_DCDM(D2)__DCC9_Slot2_GC_292_5.wav
So "TIME"_"unknown 3 numbers"_"Service method"_"Color code"_"Slot number"_"Group call"_Talk group[with or without name]"_"RID[with or without name]".wav

**Reply by noamlivne** [reply]
I see that underscores can be messed up here and converted to italics text. 
If you want me to send you the examples and my interpretation via email.

**Reply by chuot** [reply]
Thanks but they appear correctly on the email GitHub sent me.

Looking at this, I think the best way to go is to have a new `dirwatch.type=dsdplus` that will cover all cases. Your examples will help a lot, but to be sure, I think I'll have no choice but to set up my own dev box to test thoroughly.

**Reply by noamlivne** [reply]
Thank you very much!

**Comment by chuot** [comment]
You can now try the v6.5.4 which now has a dirwatch.type for DSDPlus Fast Lane. Try it and let me know,

**Reply by noamlivne** [reply]
Thank you very much! 
This will open up more listening options.

Regarding the Dirwatch and DSDPlus Fastlane:
It seems that Fastlane creates every day a different folder.
The folders are dated, so today's folder is "20220909", for example.

They are nested under "1R-Record#6", for example, (which means that this is the the sixth 1R receiver, or something like that - probably it is different from user to user).

I wonder if Rdio-Scanner can move into sub-folders according to the date. 
Ideally the Rdio-Scanner will "Dirwatch" the directory of "1R-Record#6" and will ingest all the sub-folders (that are arranged by dates):
It looks something like this:

1R-Record#6/
                      20220907/
                                      recordings...
                      20220908/
                                      recordings...
                      20220909/
                                      recordings...
If I am not clear, please tell me and I will try to send screen captures of the directories' structure.

**Reply by chuot** [reply]
With the new dirwatch type dsdplus, you just need to specify the location of the 1R-Record folder and the system ID of the system you want to ingest the files to. Note that if you are recording an NXDN network, the system ID can be derived from the network ID if not specified on the dirwatch definition. Don't worry about the subfolders by date, I use their name to get the date of the recording.

**Reply by noamlivne** [reply]
I am still at the Dirwatch stage...
Unfortunately, it does not work for me.

When I enter/change the dedicated DSDPlus Dirwatch (I have two otherDirwatches that are for SDRTrunk and work fine), ans Save the changes, the Windows program crashes.

When I start it again, I do not see that there is any indication of the new DSDPlus Dirwatch in the Windows Terminal window.

The Dirwatch is for Wav and looks something like ...\DSDPlus...\1R-Record#6
It does not ingest the Dates subfolders.

When I point it to a specific date, then it works - but only for that specific date.

Perhaps it is just me...

**Reply by chuot** [reply]
I must admit that I haven't yet tested on the windows platform as I'm currently building a test bench for that. Could you please email me a zip file with a couple of wav files (say 5) under there respective subfolders. I will check them. Thanks

**Reply by noamlivne** [reply]
Sent to your email. Thank you very much!

**Reply by chuot** [reply]
Thanks, so right away I got a crash while trying to ingest your files. So there's something different between the filename syntax of the files I tested with and the one you just sent me. Let me check what I can do to fix that.

What version of DSDPlus Fast Lane are you on?

**Reply by noamlivne** [reply]
2.368

**Comment by chuot** [comment]
Big thanks to @noamlivne for his help sorting out the DSD+ filename schemes. Users who want to give DSD+ Fast Lane a try can test with the latest v6.5.6.
